ArduCopter/mode_auto.cpp

```diff
@@ -2235,24 +2235,27 @@ bool ModeAuto::verify_nav_attitude_time(const AP_Mission::Mission_Command& cmd)
 // pause - Prevent aircraft from progressing along the track
 bool ModeAuto::pause()
 {
-    // do not pause if already paused or not in the WP sub mode or already reached to the destination
-    if(wp_nav->paused() || _mode != SubMode::WP || wp_nav->reached_wp_destination()) {
+    // do not pause if not in the WP sub mode or already reached to the destination
+    if(_mode != SubMode::WP || wp_nav->reached_wp_destination()) {
         return false;
     }
 
-    wp_nav->set_pause();
+    // do not pause if already paused
+    if (!wp_nav->paused()) {
+        wp_nav->set_pause();
+    }
+
     return true;
 }
 
 // resume - Allow aircraft to progress along the track
 bool ModeAuto::resume()
 {
     // do not resume if not paused before
-    if(!wp_nav->paused()) {
-        return false;
+    if(wp_nav->paused()) {
```

Comment:
Existing white space issue.
```suggestion
    if (wp_nav->paused()) {
```

and as I said above, looks like there is no harm in not checking and just setting.

Reply:
I agree, let's remove the check and just call wp_nav->pause().